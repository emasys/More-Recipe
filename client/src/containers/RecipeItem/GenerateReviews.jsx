import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import ReactTooltip from 'react-tooltip';

import config from '../../config';
import Auth from '../../components/auth';

const GenerateReviews = ({ review, deleteReview }) => {
  if (review.fetch_reviews) {
    return review.fetch_reviews.reviews.map((comment, index) => (
      <div
        data-aos="fade-left"
        data-aos-offset="20"
        className="comment-wrapper"
        key={comment.id}
      >
        <div className="direction mt-50 p-15 bg-light my-2">
          <div className="commentTitle ">
            <div className="float-left  mr-5">
              <img
                src={comment.User.avatar || config.DEFAULT_DISPLAY_PICTURE}
                alt="dp"
                className="img-icon-review rounded-circle"
              />
            </div>
            <div className="clearfix mb-10">
              <Link className="text-dark bolder moniker" to={`/user/${comment.userId}`}>
                {comment.User.moniker}
              </Link>
              <p className="text-dark date">
                {moment(comment.updatedAt).fromNow()}
              </p>
              {Auth.userID() === comment.userId && (
                // eslint-disable-next-line
                <i
                  onClick={() => deleteReview(comment.id, comment.recipeId)}
                  className="fa fa-times-circle delete-review-btn hvr-buzz-out fa-2x"
                  data-tip="Delete this review"
                />
              )}
              <ReactTooltip place="bottom" type="dark" effect="float" />
            </div>
            <hr />
          </div>
          <div className="comments ml-10">
            <p className="mb-0 text-dark">{comment.content}</p>
          </div>
        </div>
      </div>
    ));
  }
  return 'Loading...';
};

export default GenerateReviews;
