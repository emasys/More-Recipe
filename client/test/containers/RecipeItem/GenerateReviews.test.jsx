import React from 'react';
import { shallow, mount } from 'enzyme';
import Router from 'react-mock-router';

// Component
import GenerateReviews from '../../../src/containers/RecipeItem/GenerateReviews';
import fakeStore from '../../__mocks__/fakeStore';

let wrapper = null;

Date.now = jest.fn(() => 1487076708000);
describe('Test suite for GenerateReview component ', () => {
  describe('Render component with empty props', () => {
    const props = {
      review: {},
      deleteReview: jest.fn(),
      auth: {}
    };

    beforeEach(() => {
      wrapper = shallow(<GenerateReviews {...props} />);
    });
    it('Should render without errors', () => {
      expect(wrapper.length).toEqual(1);
      expect(wrapper.find('p').text()).toMatch('Loading comments...');
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Render component with props', () => {
    const props = {
      review: {
        review: {},
        fetch_reviews: [
          {
            id: 1,
            content: 'new comment',
            createdAt: '2018-03-15T08:03:50.868Z',
            updatedAt: '2018-03-15T08:03:50.868Z',
            recipeId: 1,
            userId: 1,
            reviewsreply: [],
            User: {
              moniker: 'admin',
              avatar: null
            }
          }
        ],
        count_reviews: 1
      },
      deleteReview: jest.fn(),
      auth: fakeStore.user
    };

    beforeEach(() => {
      wrapper = mount(<Router>
        <GenerateReviews {...props} />
      </Router>);
    });
    it('Should render without errors', () => {
      expect(wrapper.length).toEqual(1);
      expect(wrapper.find('p.mb-0.text-dark').text()).toMatch('new comment');
    });
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should delete review', () => {
      wrapper.find('.delete-review-btn').simulate('click');
      expect(props.deleteReview).toBeCalled();
    });
  });
});
