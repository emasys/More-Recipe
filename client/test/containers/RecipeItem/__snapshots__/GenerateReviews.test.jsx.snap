// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for GenerateReview component  Render component with empty props should match snapshot 1`] = `
<div>
  <p>
    Loading comments...
  </p>
</div>
`;

exports[`Test suite for GenerateReview component  Render component with props should match snapshot 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <GenerateReviews
    auth={
      Object {
        "allUsers": Object {
          "success": true,
          "users": Array [
            Object {
              "avatar": null,
              "bio": null,
              "country": null,
              "createdAt": "2018-03-13T05:04:37.443Z",
              "email": "emasysnd@gmail.com",
              "firstName": null,
              "id": 1,
              "lastName": null,
              "moniker": "admin",
              "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
              "updatedAt": "2018-03-13T05:04:37.443Z",
            },
          ],
        },
        "authInfo": Object {
          "isLoggedIn": true,
          "userId": 1,
          "username": "admin",
        },
        "isLoggedIn": true,
        "message": null,
        "userInfo": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": "Nigeria",
            "email": "emasysnd@gmail.com",
            "firstName": "emmy",
            "id": 1,
            "lastName": "endy",
            "moniker": "admin",
          },
          "success": true,
        },
        "userProfile": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": null,
            "email": "emasysnd@gmail.com",
            "firstName": null,
            "id": 1,
            "lastName": null,
            "moniker": "admin",
          },
          "success": true,
        },
      }
    }
    deleteReview={[Function]}
    review={
      Object {
        "count_reviews": 1,
        "fetch_reviews": Array [
          Object {
            "User": Object {
              "avatar": null,
              "moniker": "admin",
            },
            "content": "new comment",
            "createdAt": "2018-03-15T08:03:50.868Z",
            "id": 1,
            "recipeId": 1,
            "reviewsreply": Array [],
            "updatedAt": "2018-03-15T08:03:50.868Z",
            "userId": 1,
          },
        ],
        "review": Object {},
      }
    }
  >
    <div>
      <ReviewsList
        auth={
          Object {
            "allUsers": Object {
              "success": true,
              "users": Array [
                Object {
                  "avatar": null,
                  "bio": null,
                  "country": null,
                  "createdAt": "2018-03-13T05:04:37.443Z",
                  "email": "emasysnd@gmail.com",
                  "firstName": null,
                  "id": 1,
                  "lastName": null,
                  "moniker": "admin",
                  "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                  "updatedAt": "2018-03-13T05:04:37.443Z",
                },
              ],
            },
            "authInfo": Object {
              "isLoggedIn": true,
              "userId": 1,
              "username": "admin",
            },
            "isLoggedIn": true,
            "message": null,
            "userInfo": Object {
              "data": Object {
                "avatar": null,
                "bio": null,
                "country": "Nigeria",
                "email": "emasysnd@gmail.com",
                "firstName": "emmy",
                "id": 1,
                "lastName": "endy",
                "moniker": "admin",
              },
              "success": true,
            },
            "userProfile": Object {
              "data": Object {
                "avatar": null,
                "bio": null,
                "country": null,
                "email": "emasysnd@gmail.com",
                "firstName": null,
                "id": 1,
                "lastName": null,
                "moniker": "admin",
              },
              "success": true,
            },
          }
        }
        comment={
          Object {
            "User": Object {
              "avatar": null,
              "moniker": "admin",
            },
            "content": "new comment",
            "createdAt": "2018-03-15T08:03:50.868Z",
            "id": 1,
            "recipeId": 1,
            "reviewsreply": Array [],
            "updatedAt": "2018-03-15T08:03:50.868Z",
            "userId": 1,
          }
        }
        deleteReview={[Function]}
        key="1"
        review={
          Object {
            "count_reviews": 1,
            "fetch_reviews": Array [
              Object {
                "User": Object {
                  "avatar": null,
                  "moniker": "admin",
                },
                "content": "new comment",
                "createdAt": "2018-03-15T08:03:50.868Z",
                "id": 1,
                "recipeId": 1,
                "reviewsreply": Array [],
                "updatedAt": "2018-03-15T08:03:50.868Z",
                "userId": 1,
              },
            ],
            "review": Object {},
          }
        }
      >
        <div
          className="comment-wrapper direction"
          key="1"
        >
          <div
            data-aos="fade-left"
            data-aos-offset="20"
          >
            <div
              className=" rounded mt-50 p-15 my-2"
            >
              <div
                className="commentTitle "
              >
                <div
                  className="float-left  mr-5"
                >
                  <img
                    alt="dp"
                    className="img-icon-review rounded-circle"
                    src="https://res.cloudinary.com/emasys/image/upload/v1512284260/mejglndrmulkudy2feug.png"
                  />
                </div>
                <div
                  className="clearfix mb-10"
                >
                  <Link
                    className="text-dark bolder moniker"
                    replace={false}
                    to="/user/1"
                  >
                    <a
                      className="text-dark bolder moniker"
                      onClick={[Function]}
                    >
                      admin
                    </a>
                  </Link>
                  <p
                    className="text-dark date"
                  >
                    in a year
                  </p>
                  <i
                    className="fa fa-times-circle delete-review-btn hvr-buzz-out fa-2x "
                    data-tip="Delete this review"
                    onClick={[Function]}
                  />
                  <ReactTooltip
                    effect="float"
                    insecure={true}
                    place="bottom"
                    resizeHide={true}
                    type="dark"
                    wrapper="div"
                  >
                    <div
                      className="__react_component_tooltip place-top type-dark "
                      data-id="tooltip"
                    />
                  </ReactTooltip>
                </div>
                <hr />
              </div>
              <div
                className="comments ml-10"
              >
                <p
                  className="mb-0 text-dark"
                >
                  new comment
                </p>
              </div>
              <div
                className="reply mt-15 text-dark hovered"
              >
                <a
                  onClick={[Function]}
                >
                  Reply 
                </a>
                 |
                <a>
                   No reply 
                </a>
                <i
                  className="fa fa-chevron-down"
                />
              </div>
            </div>
          </div>
          <div
            className="container reply-container"
          />
        </div>
      </ReviewsList>
    </div>
  </GenerateReviews>
</MockRouter>
`;
